### PR TITLE
#984 Error: Couldn't acquire file lock.

### LINKED ISSUE #984: Error: Couldn't acquire file lock.
**Describe the bug**

Error message during installation of tedge-mapper-c8y:

```
2022-03-08T10:34:27.009972774Z ERROR flockfile::unix: Another instance of tedge-mapper-c8y is running.
2022-03-08T10:34:27.010008322Z ERROR flockfile::unix: Lock file path: /var/log/tedge-mapper-c8y.lock
```
Probably the directory /var/log is not the right one.

**To Reproduce**
Install tedge mapper built on top of dd607148665cbaea265211f1442c7ec8e01cc604

**Expected behavior**
Works :)

**Screenshots**

image

https://github.com/thin-edge/thin-edge.io/actions/runs/1950813139
https://github.com/abelikt/thin-edge.io/runs/5462782774?check_suite_focus=true

image

### PR DESCRIPTION
**Pull request by initard**
## Proposed changes
- Closes #984
- fixed /run path bug introduced by #939
- updated /run default configuration to use /run

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by abelikt — APPROVED** [review]
Looks good :)

File: crates/common/tedge_config/src/tedge_config.rs [hunk 1]
```diff
@@ -566,7 +566,7 @@ impl ConfigSettingAccessor<RunPathDefaultSetting> for TEdgeConfig {
             .run
             .dir_path
             .clone()
-            .unwrap_or_else(|| self.config_defaults.default_logs_path.clone()))
+            .unwrap_or_else(|| self.config_defaults.default_run_path.clone()))
```
Comment (rina23q) [inline crates/common/tedge_config/src/tedge_config.rs:569]: Flock files should be created under `/run/lock`. Doesn't this return only `/run/`?

**Review by rina23q — CHANGES_REQUESTED** [review]
Unfortunately, my question was true. I tested, then it uses `/run/tedge-mapper-c8y.lock` instead of `/run/lock/tedge-mapper-c8y.lock`.

```
$ sudo -u tedge-mapper ./tedge_mapper c8y
2022-03-08T21:46:35.325923843Z ERROR flockfile::unix: Another instance of tedge-mapper-c8y is running.
2022-03-08T21:46:35.325976232Z ERROR flockfile::unix: Lock file path: /run/tedge-mapper-c8y.lock
Error: Couldn't acquire file lock.

Caused by:
    Permission denied (os error 13)
```

File: crates/common/tedge_config/src/tedge_config.rs (hunk 1 above)
Comment (rina23q) [reply crates/common/tedge_config/src/tedge_config.rs:569]: P.S. This function should return just `/run`, this is what we ask for users. However, when creating a lock file, we have to postfix it to add `/lock`.

**Pull request merged**